### DISCUSSION TITLE
Reupload after shutting down Technichub needed or am I doing something wrong?

### DISCUSSION THREAD

**Discussion by DubbelDrankNL** [pybricks/support#1471] [General]
This is my first code: changing the colors of the Technic hub and the remote when pressing a button on the remote. It works flawlessly. When I upload a code it works.
But when I turn off the hub, the script is not responding and the LED keeps blinking blue.
Do I need to reupload the script every time the hub is powered off?

**Comment by laurensvalk** [comment]
Can you share the code that isn’t working?

Programs should be saved, but it sounds like it is not shutting down properly due to technical issue.

Your program may help us isolate and fix the issue :)

If you haven’t already, please try it again by installing the firmware from https://beta.pybricks.com/

We fixed a similar issue recently, so chances are this test version already includes a fix for your issue. Thanks!

**Comment by DubbelDrankNL** [comment]
Thank you for your quick respons!

Installing the beta firmware did resolve the problem (I think)

Here is what I do:
I upload the code
I hold the green button on the hub pressed until the LED flashes rapidly and the hub turns off.
I press the button once and the LED blinks blue
I press the button once more and the LED glows blue
I press the button the remote
Both LEDs go yellow as they should according to the code.

Maybe this is also the right procedure with the regular firmware. I will try that tomorrow.


Here is my code:

```
# Kleuren op Hub en Remote
# 2024-02-23
# Beta firmware

from pybricks.pupdevices import Remote
from pybricks.hubs import TechnicHub
from pybricks.parameters import Button, Color
from pybricks.tools import wait

# Connect to the remote.
remote = Remote()
hub = TechnicHub()

hub.light.on(Color.YELLOW)
remote.light.on(Color.YELLOW)

while True:

    # Check which buttons are pressed.
    pressed = remote.buttons.pressed()

    # Check a specific button.
    if Button.CENTER in pressed:
        hub.light.on(Color.RED)
        remote.light.on(Color.RED)
    if Button.LEFT in pressed:
        hub.light.on(Color.VIOLET)
        remote.light.on(Color.VIOLET)
    if Button.LEFT_PLUS in pressed:
        hub.light.on(Color.GREEN)
        remote.light.on(Color.GREEN)
    if Button.LEFT_MINUS in pressed:
        hub.light.on(Color.BLUE)
        remote.light.on(Color.BLUE)
     if Button.RIGHT in pressed:
        hub.light.on(Color.ORANGE)
        remote.light.on(Color.ORANGE)
    if Button.RIGHT_PLUS in pressed:
        hub.light.on(Color.MAGENTA)
        remote.light.on(Color.MAGENTA)
    if Button.RIGHT_MINUS in pressed:
        hub.light.on(Color.CYAN)
        remote.light.on(Color.CYAN)

    # Wait so we can see the result.
    # wait(100)
```

**Reply by laurensvalk** [reply]
> Installing the beta firmware did resolve the problem (I think)

Sounds good, thanks for confirming! That's another good reason to properly release the fixed firmware soon :)

**Reply by DubbelDrankNL** [reply]
With the stable firmware:
... all steps ...
Both LEDs go yellow as they should according to the code.

So it wasn't the firmware, it was me :)

**Comment by Flippoel** [comment]
Hi, it seems that we have a similar problem. Today we programmed the whole day on 4 technic hubs whom are working together with a spike prime hub as a selfmade-remote. Everything was working fine, after that we started programming another technic hub with a controller. When we want to try that, all technic hubs aren't able to run there program, only the Spike prime is still running, but all the technic hubs shut down after trying to start there program. 
We already restarted the power for the hub, we even rebooted the computer.
It's quit annoying because, you don't get any error message, so we can't figure out what's the problem.
Hopefully one of you can help us.
Thanks!

**Reply by dlech** [reply]
Can you please start a new discussion and share the programs and exact firmware version you are using so we can try to reproduce the problem?

**Reply by Flippoel** [reply]
Hi David, thanks for your reply. I hope to start a new discusion today. 
One question on forehand, how do I get the firmware version out of the hubs?

**Reply by laurensvalk** [reply]
Press the >>> button next to the stop button. It will start the interactive prompt, and it will display the version too. The you can just press stop again since you don't need the prompt.

**Reply by laurensvalk** [reply]
As explained in the post above, it's best to make a simple test program that only turns on the light for a few seconds. See if you can save that program successfully before trying something big.
